dotCMS/src/main/java/com/dotcms/security/apps/AppsUtil.java

```diff
@@ -217,6 +272,219 @@ public static String digest(final String text) {
     }
 
 
+    /**
+     * Loads the default password stored in the properties
+     */
+    public static String loadPass(final Supplier<String> passwordOverride) {
+        String password = null;
+        if (null != passwordOverride) {
+            Logger.info(AppsUtil.class,"Apps Password Override supplier has been provided.");
+            password = passwordOverride.get();
+        }
+        if(isNotSet(password)) {
+            Logger.info(AppsUtil.class,"Apps Password default will be used.");
+            password = Config.getStringProperty(APPS_IMPORT_EXPORT_DEFAULT_PASSWORD);
+        }
+        return password;
+    }
+
+    /**
+     * Ths will return a list of AppSecrets arranged in a map whose key is the site-id
+     * Therefore we have a Secrets by site
+     * @param incomingFile
+     * @param key
+     * @return
+     * @throws DotDataException
+     * @throws IOException
+     * @throws EncryptorException
+     */
+
+     public static Map<String, List<AppSecrets>> importSecrets(final Path incomingFile, final Key key)
+            throws DotDataException, IOException {
+
+        final byte[] encryptedBytes = Files.readAllBytes(incomingFile);
+        final byte[] decryptedBytes;
+        try {
+            decryptedBytes = AppsUtil.decrypt(key, encryptedBytes);
+        }catch (EncryptorException e){
+            throw new IllegalArgumentException("An error occurred while decrypting file contents. ",e.getCause());
```

Comment:
Codacy found an issue: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)